libraries/_helper.rb

```diff
@@ -291,9 +297,11 @@ def delivery_server_node
 
     def delivery_server_fqdn
       @delivery_server_fqdn ||= begin
-        delivery_server_fqdn   = get_ip(delivery_server_node)
+        delivery_server_fqdn  = get_ip(delivery_server_node)
         Chef::Log.info("Your Delivery Server Public/Private IP is => #{delivery_server_fqdn}")
-        node['delivery-cluster']['delivery']['fqdn'] || delivery_server_fqdn
+        node['delivery-cluster']['delivery']['fqdn'] ||
+        node['delivery-cluster']['delivery']['host'] ||
+        delivery_server_fqdn
```

Comment:
Might be time for a helper function that takes the component as an argument? @afiune